zerver/webhooks/circleci/view.py

```diff
@@ -23,133 +48,98 @@ def api_circleci_webhook(
     user_profile: UserProfile,
     payload: WildValue = REQ(argument_type="body", converter=to_wild_value),
 ) -> HttpResponse:
-    payload = payload["payload"]
     subject = get_subject(payload)
     body = get_body(payload)
 
+    pipeline = payload["pipeline"]
+
+    # We currently don't support projects using VCS providers other than GitHub,
+    # BitBucket and GitLab.
+    if "trigger_parameters" in pipeline and pipeline["trigger"]["type"] != "gitlab":
```

Comment:
I don't understand why it's only checking for `gitlab`, and not the other two providers, here.

Reply:
I guess the others don't use this `trigger_parameters` format?